### ISSUE TITLE
Is the upper limit of the LTX-Video's effect is too low?

### ISSUE THREAD

**Issue by 2502128021** [eisneim/ltx_lora_training_i2v_t2v#8]
Judging from the actual test performance of the LTX-Video and the demo samples you provided, it seems the upper limit of the model is low? For example, the video quality, textures of objects, the teeth of people are always terrible.
Do you have any idea for imporving the performance of LTX-Video? or just choose other models like HunYuan-Video.

**Comment by eisneim** [comment]
today i'm working on combining [Enhance-A-Video](https://github.com/NUS-HPC-AI-Lab/Enhance-A-Video) with STG guidance, try to improve output quality of LTX, Hunyuan or Mochi is better of course but we don't have 80Gb H100 GPU and don't want to wait 10 minutes on RTX 4090, so i'm still trying my best luck with LTX; 

LTX team only release the 2B model, they have a bigger version internally that is not released, i guess the 5B version of LTX is going to be the best choice for consumer level GPU that banlance quality and speed;

**Comment by 2502128021** [comment]
I‘m trying to add additional cross-attn layer to support extra control after I trained Lora, but the training loss degrades so slowly. The training speed of video generation model is so slow and I'm not sure is it possible to train the extra cross-attn layers well.

**Comment by eisneim** [comment]
@2502128021 “so slow” what GPU? i use 2 RTX 4090 48GB to train different loras, for batch size of 3 rank=256 all layer cost about 6s/it while the GPU is still doing other precompute, if no other jobs running i think it should be 3s/it ;  after running for 24 hours the lora is good already.

i'm training a new lora [Cinemagraph](https://civitai.com/models/1340555/ltxcinemagraphnobodymoves) with batch size 3 cost about 48 hours

https://github.com/user-attachments/assets/175d3d8c-af6b-4987-9ad0-af6584ab66aa

**Comment by 2502128021** [comment]
I'm conducting training on the L20 with 45G of video memory. In the first stage, I trained a LoRA for character movement and speech, and the effect basically met the target. In the second stage, based on loading the already trained LoRA, I attempted to train the additionally added cross-attention layers for conditional control. However, the effect is emerging very slowly. It seems that after several days of training, the conditional control still doesn't seem to be taking effect.

**Comment by eisneim** [comment]
you can fuse the lora layers and output a new model weights, then train a new cross attention layers based on the new weight, train with locked weights for 1 day, then, unlock all layers  for 1 day, see if this method works

**Comment by 2502128021** [comment]
This sounds quite reasonable. Additionally, it seems that both the resolution and duration of my training are a bit large. I trained with a resolution of 1280 * 736 over 257 frames, which might be one of the reasons why the training effect hasn't shown up yet. I'm really curious as to why the generation length of current video generation models is so short. How can long videos be generated?

**Comment by eisneim** [comment]
diffusion models are not like Autoregressive LLMs they generate fixed sequence length, if the base model is not trained with long seq not the finetune is going to be much harder; longer sequence is also harder for diffusion model, basically you are making the noise guessing problem harder; 

for longer generation you can try: [FlexiTex](https://flexitex.github.io/FlexiTex/)

or another very promising method [diffusion forcing](https://github.com/kwsong0113/diffusion-forcing-transformer)

**Comment by 2502128021** [comment]
Thank you for sharing. I'm amazed that training a LoRA for a resolution of 1280x736 over 257 frames is feasible. It seems that the model can normally generate videos of this length. Considering that the code uses 3D RoPE for positional encoding, I suspect that if there is enough VRAM, it might be possible to train for even longer video sequences. However, it seems that the LTX-Video base model itself is not sufficiently trained for this resolution and length. Additionally, the VAE part of LTX-Video is quite poor. I conducted reconstruction experiments and found that after encoding and decoding, there are noticeable quality drops in areas such as the eyes and mouth of the people in the video. I feel that if the VAE could be improved, the quality of the generated videos would likely be much better.

**Comment by eisneim** [comment]
the VAE in LTX has very very high compression compared to Cogv, Hunyuan, it has 32x32 for spatial and 8x for temporal that's why the reconstruction is not very good especially for high frequency part; LTX use RoPE theta=10000 so it should be able to handle long sequence; we can chat throw wechat if you have one (id: eisneim)
